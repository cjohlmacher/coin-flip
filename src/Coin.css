.Coin {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    margin: 2px;
}

.Coin-Image {
    height: 50vh;
    width: auto;
}