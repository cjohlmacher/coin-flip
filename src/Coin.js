import React from 'react';
import './Coin.css';

const Coin = ({isHeads}) => {
    
    const headsImg = <img className="Coin-Image" alt="coin-heads" src="https://images.pexels.com/photos/7702774/pexels-photo-7702774.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500" />;
    const tailsImg = <img className="Coin-Image" alt="coin-tails" src="https://images.pexels.com/photos/8466440/pexels-photo-8466440.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500" />;
    const img = isHeads ? headsImg : tailsImg;

    return (
        <div className="Coin">
            {img}
        </div>
    )
};

export default Coin;