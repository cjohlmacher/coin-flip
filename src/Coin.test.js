import { render } from '@testing-library/react';
import Coin from './Coin';

it('renders without crashing', () => {
    render(<Coin isHeads={true}/>);
});

it('matches snapshot when heads', () => {
    const {asFragment} = render(<Coin isHeads={true} />);
    expect(asFragment()).toMatchSnapshot();
});

it('matches snapshot when tails', () => {
    const {asFragment} = render(<Coin isHeads={false} />);
    expect(asFragment()).toMatchSnapshot();
});