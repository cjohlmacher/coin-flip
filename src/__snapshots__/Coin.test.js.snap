// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when heads 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    <img
      alt="coin-heads"
      class="Coin-Image"
      src="https://images.pexels.com/photos/7702774/pexels-photo-7702774.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when tails 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    <img
      alt="coin-tails"
      class="Coin-Image"
      src="https://images.pexels.com/photos/8466440/pexels-photo-8466440.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
    />
  </div>
</DocumentFragment>
`;
